docs/source/EccFramework.rst

```diff
@@ -0,0 +1,201 @@
+ECC Framework: Automatic Implementation and Evaluation of Error-Correcting Codes for Quantum Computing
+======================================================================================================
+
+The QECC library offers means for automatic implementation and evaluation of error-correcting codes for quantum computing. More precisely, the library allows to automatically apply different error correction schemes to quantum circuits provided as openqasm files. The "protected" quantum circuits can then be exported again in the form of openqasm files or can be directly used for noise-aware quantum circuit simulation. For the latter case, we also provide a wrapper script which makes use of the provided framework to apply error correction schemes to circuits and directly simulate those circuits using qiskit.
+
+Installation
+############
+
+Installation for Users
+----------
+
+.. code-block:: console
+
+  (venv) $ pip install mqt.qecc
+
+
+Installation for Developers
+----------
+
+If you have not done so already, clone the repository using:
+
+.. code-block:: console
+
+  git clone --recurse-submodules https://github.com/cda-tum/qecc
+
+Make sure you are in the main project directory for the next steps.
+
+.. code-block:: console
+
+  cd qecc
+
+and (if necessary), update the submodules.
+
+.. code-block:: console
+
+  git submodule update --init --recursive
+
+Then, the ECC framework can be installed using pip
+
+.. code-block:: console
+
+  (venv) pip install --editable .
+
+If you want to use Qiskit for quantum circuit simulation, you need to install it as well
+
+.. code-block:: console
+
+  (venv) pip install qiskit
+
+Usage
+#####
+
+Having the Python module installed, error correcting codes can be applied using apply_ecc of module qecc, like so
+
+.. code-block:: python
+
+  from mqt import qecc
+
+  file = "path/to/qasm/file.qasm"  # Path to the openqasm file the quantum circuit shall be loaded from
+  ecc = "Q7Steane"  # Error correction code that shall be applied to the quantum circuit
+  ecc_frequency = 100  # After how many times a qubit is used, error correction is applied
+
+  result = qecc.apply_ecc(file, ecc, ecc_frequency)
+
+  # print the resulting circuit
```

Comment:
This comment is superfluous as it just spells out what the code down below says anyways